src/System.Linq/src/System/Linq/ToCollection.cs

```diff
@@ -176,5 +176,18 @@ public static List<TSource> ToList<TSource>(this IEnumerable<TSource> source)
 
             return d;
         }
+
+        public static HashSet<TElement> ToHashSet<TElement>(this IEnumerable<TElement> source) => source.ToHashSet(null);
```

Comment:
Should the generic parameter be named `TSource` to match the naming used by `ToList`, `ToArray`, and `ToDictionary`? `TElement` is only used for the `ToDictionary` overloads with an `elementSelector`.